service/src/test/java/bio/terra/workspace/service/privateresource/PrivateResourceCleanupServiceTest.java

```diff
@@ -0,0 +1,337 @@
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledIfEnvironmentVariable;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ActiveProfiles;
+
+// Use application configuration profile in addition to the standard connected test profile.
+@ActiveProfiles({"connected-test", "configuration-test"})
+public class PrivateResourceCleanupServiceTest extends BaseConnectedTest {
+
+  // The name of the "member" policy created by default for groups. If this is ever used for
+  // implementation code, it should live in SamConstants instead.
+  private static final String SAM_GROUP_MEMBER_POLICY = "member";
+  private Workspace workspace;
+  private GroupApi ownerGroupApi;
+  private String groupName;
+  private String groupEmail;
+
+  @Autowired WorkspaceConnectedTestUtils workspaceUtils;
+  @Autowired UserAccessUtils userAccessUtils;
+  @Autowired ApplicationAccessUtils applicationAccessUtils;
+  @Autowired SamConfiguration samConfiguration;
+  @Autowired WorkspaceService workspaceService;
+  @Autowired SamService samService;
+  @Autowired ControlledResourceService controlledResourceService;
+  @Autowired PrivateResourceCleanupConfiguration privateResourceCleanupConfiguration;
+  @Autowired PrivateResourceCleanupService privateResourceCleanupService;
+  @Autowired ResourceDao resourceDao;
+  @Autowired WsmApplicationConfiguration appConfig;
+  @Autowired WsmApplicationService wsmApplicationService;
+
+  /** Set up default workspace, group ID, and GroupApi client object. */
+  @BeforeEach
+  public void setup() {
+    // The cleanup cronjob will not run because it's initially disabled via configuration, but for
+    // these tests we can enable it and manually trigger cleanup. No gap between runs to ensure
+    // tests don't interfere with each other.
+    privateResourceCleanupConfiguration.setEnabled(true);
+    privateResourceCleanupConfiguration.setPollingInterval(Duration.ZERO);
+    // Client API objects manage their own thread pools. If we add more tests, we should consider
+    // factoring this into a shared object across tests.
+    ownerGroupApi = buildGroupApi(userAccessUtils.defaultUserAuthRequest());
+    workspace =
+        workspaceUtils.createWorkspaceWithGcpContext(userAccessUtils.defaultUserAuthRequest());
+    groupName = UUID.randomUUID().toString();
+    groupEmail = createGroup(groupName, ownerGroupApi);
+  }
+
+  /**
+   * Delete workspace. Doing this outside of test bodies ensures cleanup runs even if tests fail.
+   */
+  @AfterEach
+  private void cleanup() {
+    workspaceService.deleteWorkspace(
+        workspace.getWorkspaceId(), userAccessUtils.defaultUserAuthRequest());
+    deleteGroup(groupName, ownerGroupApi);
+  }
+
+  @Test
+  @DisabledIfEnvironmentVariable(named = "TEST_ENV", matches = BUFFER_SERVICE_DISABLED_ENVS_REG_EX)
```

Comment:
just curious, why is this disabled for TEST_ENV while the other method doesn't need to?